Python/flowgraph.c

```diff
@@ -1771,13 +1809,13 @@ optimize_cfg(cfg_builder *g, PyObject *consts, PyObject *const_cache, int firstl
     }
     RETURN_IF_ERROR(remove_unreachable(g->g_entryblock));
 
-    for (int n = 0; n < 2; n++) {
-        for (basicblock *b = g->g_entryblock; b != NULL; b = b->b_next) {
-            remove_redundant_nops(b);
-        }
-        RETURN_IF_ERROR(remove_redundant_jumps(g));
-    }
-
+    int removed_nops, removed_jumps;
+    do {
+        removed_nops = remove_redundant_nops(g);
+        RETURN_IF_ERROR(removed_nops);
+        removed_jumps = remove_redundant_jumps(g);
+        RETURN_IF_ERROR(removed_jumps);
+    } while(removed_nops + removed_jumps > 0);
```

Comment:
Could you add a comment about why this is guaranteed to terminate.
I presume it is guaranteed, as both `remove_redundant_nops` and `remove_redundant_jumps` monotonically reduce the number of instructions?

Reply:
remove_redundant_jumps may not reduce the number of instructions - it may replace a jump by a NOP.